src/store/formStore.ts

```diff
@@ -0,0 +1,41 @@
+import { create } from 'zustand';
+import { countries as allCountries } from '../utils/countries';
+import type { FormSchema } from '../schemas/formSchema';
+
+export type Store = {
+  controlledForms: (FormSchema & { _new?: boolean })[];
+  uncontrolledForms: (FormSchema & { _new?: boolean })[];
+  countries: string[];
+  addControlledForm: (data: FormSchema) => void;
+  addUncontrolledForm: (data: FormSchema) => void;
+  hydrateCountries: () => void;
+};
+
+export const useFormStore = create<Store>((set) => ({
+  controlledForms: [],
+  uncontrolledForms: [],
+  countries: [],
+  addControlledForm: (data) => {
+    set((s) => ({
+      controlledForms: [...s.controlledForms, { ...data, _new: true }],
+    }));
+    setTimeout(() => {
+      set((s) => ({
+        controlledForms: s.controlledForms.map((x) => ({ ...x, _new: false })),
+      }));
+    }, 2500);
+  },
```

Comment:
The logic in lines `19–26` and `29–36` is very similar, so you can create a helper function like addFormWithFlag to encapsulate it. Then, the store actions can delegate to this helper for cleaner and more maintainable code, for example:
```
  addControlledForm: (data) => {
    addFormWithFlag(set, 'controlledForms', data);
  },
  addUncontrolledForm: (data) => {
    addFormWithFlag(set, 'uncontrolledForms', data);
  },
```